### PR TITLE
updated outliers.zscore to include nan_policy

### LINKED ISSUE #102: Handling missing data in outliers
Hello, first of all very interesting project!

I was trying out the outlier funcions. The `outliers.zscore` function (which depends on `scipy.stats.zscore`) returns a series full of NaNs when at least one NaN is present in data.

Would it be helpful to use the option of ignoring such data? Like `stats.zscore(data, nan_policy='omit')` or dropping NaN values?
Or handling this type of data is not to be contemplated?

### PR DESCRIPTION
**Pull request by MichaelHopwood**
## Description

An update to `outliers.zscore` to expose the nan_policy.

## Checklist

The following items must be addressed before the code can be merged. 
Please don't hesitate to ask for help if you are unsure of how to accomplish any of the items. 
*You are free to remove any checklist items that do not apply or add additional items that are 
not on this list*

- [x] Closes #102
- [ ] Added new API functions to `docs/api.rst`
- [ ] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings
- [x] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- [ ] Non-API functions clearly documented with docstrings or comments as necessary
- [x] Added tests to cover all new or modified code
- [x] Pull request is nearly complete and ready for detailed review

### CODE REVIEW

**Comment by wfvining** [comment]
@MichaelHopwood I enabled the CI pipeline for you, sorry about the delay, I think this is a new feature of Github Actions.

**Comment by MichaelHopwood** [comment]
Thanks @wfvining! I believe I fixed the minor linting issues but will need to confirm once the CI pipeline is activated again.

**Comment by wfvining** [comment]
@MichaelHopwood sorry about that... this is a very inconvenient "feature." I'll keep a close eye to try to approve runs quickly. Apparently people were using actions for crypto mining, so there isn't a way to just grant blanked approval.

I would recommend running `pytest pvanalytics` and `flake8` locally to speed up the feedback loop for yourself for the time being.

Once this PR is merged any future contributions should not require maintainer approval for the actions pipeline to run.

**Comment by MichaelHopwood** [comment]
Hi @wfvining, `scipy.stats.zscore` was given the parameter `nan_policy` in release 1.4.0 [1]. For this PR to run successfully, the requirements files would have to reflect the newer version of scipy. I'll leave it up to you to decide. 👍 

[1] http://scipy.github.io/devdocs/release.1.4.0.html#scipy-stats-improvements

**Comment by wfvining** [comment]
Shoot, I did not realize that parameter was added so recently. We've been trying to apply the NEP-29 rules for dependencies, so that means we should not require scipy > 1.2.X yet. 1.4 isn't that new, but still withing 24 months (December, 2019) and 3 minor versions.

It would not be too difficult to mimic the scipy `raise` behavior, the other options look somewhat more complex, and I'm not sure we want to got to the effort of recreating them. Maybe just `raise` is enough, with a useful message that indicates users should get rid of their nans. What do you think?

**Comment by wfvining** [comment]
Another option would be to try passing `nan_policy` and catch the `TypeError` if the older version of scipy is installed. In that case, we would revert to raising our own error if there are nans, or calling `zscore` without the `nan_policy` arg if there are not. We'd probably need to also raise a `ValueError` if the user passes something other than `'raise'` in this case, something that says to upgrade to scipy >= 1.4 if they need that behavior.

**Comment by cwhanse** [comment]
If we're not advancing `scipy` then I would favor mimicking the `raise` and `omit` behavior. I don't think that `propagate` makes much sense here (any `nan` in the input makes the output `nan`). If `raise`, any `nan` in input raises a `ValueError`; if `omit`, the zscore  is returned in position where data is not `nan`, along with `nan` in positions corresponding to input.

**Comment by wfvining** [comment]
`propagate` is the behavior prior to 1.4, so we kind of get that for free. I looked at how scipy handles `omit` and it is a bit more involved than I expected. Maybe for our purposes it wouldn't be so complex: Just drop the nans, compute the zscores, reindex and restore the nans?

**Comment by cwhanse** [comment]
> Just drop the nans, compute the zscores, reindex and restore the nans?

That's what I am thinking.

**Comment by MichaelHopwood** [comment]
Thanks for the feedback! I think I captured the verdict regarding NaN handling.

File: pvanalytics/quality/outliers.py
Comment (wfvining) [inline pvanalytics/quality/outliers.py:52]:
```suggestion
    nan_policy : str, default 'raise'
```

File: pvanalytics/quality/outliers.py
Comment (wfvining) [inline pvanalytics/quality/outliers.py:56]:
```suggestion
        If 'raise', a ValueError is raised when `data` contains NaNs.
        If 'omit', NaNs are ignored and False is returned at indices that
        contained NaN in `data`.
```

File: pvanalytics/quality/outliers.py
Comment (wfvining) [inline pvanalytics/quality/outliers.py:68]:
```suggestion
    if data.hasnans:
```

File: pvanalytics/quality/outliers.py
Comment (wfvining) [inline pvanalytics/quality/outliers.py:75]: Add the valid options to the error message, and remove the name of the function (it will already be visible in the stack trace).

Also, don't need `+` for strings, python will concatenate adjacent strings automatically in this context:
```suggestion
            raise ValueError("Incorrect specification passed "
                             "to zscore's nan_policy.")
```

File: pvanalytics/tests/quality/test_outliers.py
Comment (wfvining) [inline pvanalytics/tests/quality/test_outliers.py:54]: Take a look at [`pytest.raises()`](https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions)

File: pvanalytics/quality/outliers.py
Comment (wfvining) [reply pvanalytics/quality/outliers.py]: > Add the valid options to the error message, and remove the name of the function.

Can you add this as well? It really makes a difference for users to be able to address errors quickly.

Comment (MichaelHopwood) [reply pvanalytics/quality/outliers.py]: Incorporated! Thanks!

File: pvanalytics/quality/outliers.py
Comment (wfvining) [inline pvanalytics/quality/outliers.py:75]:
```suggestion
            raise ValueError(f"Unnexpected value ({nan_policy}) passed to nan_policy. "
                             "Expected 'raise' or 'omit'.")
```

File: pvanalytics/tests/quality/test_outliers.py
Comment (cwhanse) [reply pvanalytics/tests/quality/test_outliers.py]: @MichaelHopwood suggest

```
with pytest.raises(ValueError, match='Incorrect specification passed'):
    outliers.zscore(data, nan_policy='raise')
```

**Review by cwhanse — APPROVED** [review]

**Review by wfvining — COMMENTED** [review]
This is my last nit to pick on error messages, I swear.

**Comment by wfvining** [comment]
Just needs coverage for the unnexpected nan_policy value. Then good-to-go.

**Review by wfvining — APPROVED** [review]

**Pull request merged**
